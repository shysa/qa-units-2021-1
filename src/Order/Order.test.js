import React from 'react'
import Order from "./Order";
import {fakeOrders} from "../data/fakeOrders";
import {shallow, configure} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

import {getDate} from "../utils/getDate";
jest.mock('../utils/getDate');

configure({ adapter: new Adapter() });

describe('Order.js', () => {
  beforeAll(() => {
    getDate.mockReturnValue('11 марта, чт, 2021 год');
  })

  afterAll(() => {
    getDate.mockClear();
  })

  it('render correctly component', () => {
    const testComponent = shallow(<Order order={fakeOrders[0]}/>);
    expect(testComponent).toMatchSnapshot();
  })

  it('check null order', () => {
    const order = null;
    const testComponent = shallow(<Order order={order} />);

    expect(testComponent).toEqual({});
  });

  it('check no items in order', () => {
    const order = {
      shop: 'test shop',
      date: '11 марта, чт, 2021 год',
      items: null
    }
    const testComponent = shallow(<Order order={order} />);

    expect(testComponent).toMatchSnapshot();
  });

  it('check null shop', () => {
    const order = {
      shop: null,
      date: '11 марта, чт, 2021 год',
    }
    const testComponent = shallow(<Order order={order} />);

    expect(testComponent).toEqual({});
  });

  it('check null date', () => {
    const order = {
      shop: 'test shop',
      date: null,
    }
    const testComponent = shallow(<Order order={order} />);

    expect(testComponent).toEqual({});
  });
})


