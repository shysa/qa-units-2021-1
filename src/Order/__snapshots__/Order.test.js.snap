// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js check no items in order 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      test shop
    </span>
    <span
      id="date"
    >
      11 марта, чт, 2021 год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js render correctly component 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      11 марта, чт, 2021 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;
